### PR TITLE
Setup Continuous deployment to Production

### PR DESCRIPTION
**Pull request by arku**
Fixes https://github.com/TheOdinProject/theodinproject/issues/1112

Ref: https://docs.travis-ci.com/user/deployment-v2/providers/heroku/

Whenever a build on `master` completes successfully, the code will be deployed to Production.

Note that this means once a PR is merged to `master`, it will be deployed. If we want to have control over the Production deployment, we could add a new branch `develop` and have all PRs submitted against `develop`. Once we want to deploy the changes to Production, we can cherry-pick the commits from `develop` to `master` and push to trigger a build.

This PR also fixes the [broken bundle audit check](https://travis-ci.org/github/TheOdinProject/theodinproject/builds/678347822#L802) on Travis.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,10 +12,10 @@
     "stickyfilljs": "^2.1.0",
     "toaster": "^0.1.2",
     "turbolinks": "^5.2.0",
-    "validate.js": "^0.13.1"
+    "validate.js": "^0.13.1",
+    "babel-plugin-prismjs": "^2.0.1"
```
Comment (arku) [inline package.json:16]: @TheOdinProject/odinmods  This is ugly 🙈 but not sure why the Heroku ruby buildpack does not install both `dependencies` and `devDependencies` as [Heroku Node.js buildpack](https://devcenter.heroku.com/articles/nodejs-support#package-installation) does.

If anyone knows a better solution, do let me know!

Comment (KevinMulhern) [reply package.json:16]: Can you put this fix in a separate PR please Arun? We can't deploy without it.

**Comment by KevinMulhern** [comment]
I have a few questions about this @arku, it would probably be better to have a discussion on discord instead of going back on forth on this PR though.

File: package.json (hunk 1 above)
Comment (arku) [reply package.json:16]: Sure.

Comment (arku) [reply package.json:16]: Just opened https://github.com/TheOdinProject/theodinproject/pull/1123 addressing this.

File: .travis.yml [hunk 2]
```diff
@@ -8,9 +8,25 @@ before_script:
...
+deploy:
+  - &deploy
+    provider: heroku
+    strategy: git
+    edge: true
+    api_key:
+      secure: mQVwLlrNOjZKdoI4TlZAiWayX7FFc8bI9yglHxa1L7LMQUWKQdJIduCe78wHCAYc/vyu896+N9PIXSxoU6GHs1JomMEC7mxHZOrjyY6XIZQL4tAi2DHsKO1AkF4flTovNXZH/khn5XGLHkHgDNotYglHBsRkwKC7Gu/1s5rmRup5A2X9tvMfHJ8kFCK3DK0qoP6ZFxLR9FvBfEzc+erP1iinCcjPsf2eFrlUC9NJWVk6Thf0fXF185DkXs4F/te0DwHPrrzzw5Nb4fzX+id6kPmzffW0bTAKqIuNSO79Sau8t07w/zxacZk2QsGgPKPLgEj4oIY+afbM9VlETKL+KHwHDXlrYef0Wdrj60MQEpsnbl5EUvTmO+6Cmugf51mBjHbYRttFLTHF+4vEXg2NLIn+vtHnS43QA05OxIq0LyhP2pGJz1moE05bw9mksCQKPbs5e1VE2DEN8YN+SqKC8lDeoFxeyIHT6o6CZVdubmNXXLomhgyCbvIlYABc2aTgSs8NPw7Qum8TgNeHsBBqEzMXITpgWb/g5chF3V6n5JSZn/kZ42Fn09RyeN7TjwTUH7Xu69oTagfLLdL5p2LalzpYOJAdZ0ygDSSu7vj7aFIHEilNYZc5EHPISRVFhVi3P+M/bHrY/w1U0ZEQMeBlqwi/y8ZjnGcz3oEpHXuyrQw=
+    run:
+      # Ref: https://docs.travis-ci.com/user/deployment-v2/providers/heroku/#running-commands
+      - sleep 2
```
Comment (arku) [inline .travis.yml:24]: The 2 seconds is chosen arbitrarily. Travis doesn't have a guideline on the sleep timing. We'll have to update if we find frequent build failures when trying to run `rails db:migrate`.

File: package.json (hunk 1 above)
Comment (KevinMulhern) [reply package.json:16]: Thanks @arku 💪

File: .travis.yml (hunk 2 above)
Comment (KevinMulhern) [reply .travis.yml:24]: We might be able to remove db:migrate from this script. I added a heroku release phase for migrating the db after a deploy earlier.
https://github.com/TheOdinProject/theodinproject/blob/master/Procfile

Comment (arku) [reply .travis.yml:24]: Nice!

**Comment by arku** [comment]
I have updated the PR to only deploy to Production on pushes to the `master` branch. We will be using https://devcenter.heroku.com/articles/github-integration-review-apps to review PRs.

**Comment by arku** [comment]
I have updated the PR to only deploy to Production on pushes to the `master` branch. We will be using https://devcenter.heroku.com/articles/github-integration-review-apps to review PRs.

**Pull request merged**